sound/soc/sof/intel/intel-ipc.c

```diff
@@ -73,6 +73,13 @@ int intel_pcm_open(struct snd_sof_dev *sdev,
 	/* binding pcm substream to hda stream */
 	substream->runtime->private_data = stream;
 
+	/* minimum as per HDA spec */
+	snd_pcm_hw_constraint_step(substream->runtime, 0, SNDRV_PCM_HW_PARAM_PERIOD_BYTES, 4);
+
+	/* avoid circular buffer wrap in middle of period */
+	snd_pcm_hw_constraint_integer(substream->runtime,
+				      SNDRV_PCM_HW_PARAM_PERIODS);
+
```

Comment:
why do we have the same stuff in two places? This doesn't seem to be related to HDAudio?

Also we're discussing in https://github.com/thesofproject/sof/issues/3503 and https://github.com/thesofproject/sof/issues/3492 that Broadwell seems to have more restrictions on the DMA than BYT/CHT, so we might need to make this really platform-specific.

Reply:
BYT/CHT/etal use intel_pcm_open() while APL and newer call hda_dsp_pcm_open(). As it indeed sounds like BYT/CHT might have even more constraints, it seemed best to keep these separate and have a clear place where to add the constraints.

The integer constraint is not HDA specific, but it's something we have applied for Intel hw in the past and we have at least some Intel platforms where this is likely mandatory. I still don't see a need to make this constraint in common SOF code as that will limit all platforms.

Reply:
you may want to remove the reference to HDAudio in the non-HDaudio case then